core/src/lib/StitchAppClientConfiguration.ts

```diff
@@ -1,9 +1,37 @@
-import StitchClientConfiguration from "./StitchClientConfiguration";
+import { StitchClientConfiguration } from "./StitchClientConfiguration";
 
-export default class StitchAppClientConfiguration extends StitchClientConfiguration {
-  public static Builder = class extends StitchClientConfiguration.Builder {
+export class StitchAppClientConfiguration extends StitchClientConfiguration {
+  public readonly clientAppId: string;
+  public readonly localAppName: string;
+  public readonly localAppVersion: string;
+
+  public constructor(
+    config: StitchClientConfiguration,
+    clientAppId: string,
+    localAppName: string,
+    localAppVersion: string
+  ) {
+    super(
+      config.baseURL,
+      config.storage,
+      config.dataDirectory,
+      config.transport
+    );
+    this.clientAppId = clientAppId;
+    this.localAppVersion = localAppVersion;
+    this.localAppName = localAppName;
+  }
+
+  public builder(): StitchAppClientConfiguration.Builder {
```

Comment:
The main reason I had to declare the inner classes in this way is so that these `builder()` functions could return the proper type, which they were previously not doing. Once I gave these functions return types, the TS compiler caught a lot of issues.